engine/class_modules/sc_death_knight.cpp

```diff
@@ -4450,6 +4457,7 @@ struct empower_rune_weapon_t : public death_knight_spell_t
     dot_duration = base_tick_time = 0_ms;
 
     cooldown -> duration *= 1.0 + p -> vision_of_perfection_minor_cdr;
+    cooldown -> adjust( timespan_t::from_seconds( abs( p -> spec.empower_rune_weapon_rank_2->effectN(1).base_value() / 1000 ) ) );
```

Comment:
adjust() dynamically adjusts the remaining duration on a cooldown object, the only thing you need here is to change cooldown -> duration like the line above.